### PR TITLE
Changes to add NOAA-21 and GOES-18 radiances in monitor mode

### PR DESCRIPTION
**Pull request by ADCollard**
Turn on NOAA-21 and GOES-19 radiances in operations in monitoring mode.

scripts/exglobal_atmos_analysis.sh: Add processing of GOES-18 CSRs, NOAA-21 ATMS and CrIS
fix/global_anavinfo.l127.txt: Point to new file, Rcov_crisn21
fix/global_satinfo.txt:  Down weighted NOAA-21 CrIS radiances relative to NOAA-20
fix/Rcov_crisn21: New file

The fix branch is DA_GFSv16.3.9_global_only

**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: scripts/exglobal_atmos_analysis.sh
Comment (emilyhcliu) [inline scripts/exglobal_atmos_analysis.sh:907]: Will we assimilate ABI g18 clear-sky radiance?
Or, this is a place holder?

File: scripts/exglobal_atmos_analysis.sh
Comment (emilyhcliu) [inline scripts/exglobal_atmos_analysis.sh:922]: ATMS N21 will be assimilated and the CrIS N21 is a place holder, correct?

Comment (emilyhcliu) [reply scripts/exglobal_atmos_analysis.sh:922]: I can not see the fix files from the github page.  I believe that the use flag for ATMS N21 is set to 1, and the use flags for CrIS-fsr N21 and ABI-g18 are set to -1

I think we already made changes in abias and abias_pc for the three sensors in  v16.3. implementation.

Comment (RussTreadon-NOAA) [reply scripts/exglobal_atmos_analysis.sh:922]: @emilyhcliu, I cloned `ADCollard:release/gfsda.v16`.   A check of `fix/global_satinfo.txt` shows that the usage flags for atms_n21, cris-fsr_n21, and abi_g18 are -1 for all channels.

@ADCollard , the PR description indicates that we should see changes in `fix/global_anavinfo.l127.txt`.   This file is identical between `ADCollard:release/gfsda.v16` and `NOAA-EMC:release/gfsda.v16`.   

Should we also see a difference between cris-fsr_n21 weights when comparing `global_satinfo.txt` from `ADCollard:release/gfsda.v16` and `NOAA-EMC:release/gfsda.v16`?

Comment (emilyhcliu) [reply scripts/exglobal_atmos_analysis.sh:922]: Thanks @RussTreadon-NOAA 
@ADCollard did mention that the purpose of this PR is to set CrIS N21, ATMS N21, and ABI G18 to monitoring mode in the description section of this PR.  
So, the use flags for these three sensors should be -1 and they are set correctly in this PR.

Comment (RussTreadon-NOAA) [reply scripts/exglobal_atmos_analysis.sh:922]: Yes, both current operations, gfs.v16.3.9, and this PR maintain all channels of atms_n21, cris-fsr_n21, and abi_g18 in monitor mode.   

The only difference between `ADCollard:release/gfsda.v16/fix` and `/lfs/h1/ops/prod/packages/gfs.v16.3.9/fix/fix_gsi` is the addition of `Rcov_crisn21` in `ADCollard:release/gfsda.v16/fix`

Comment (ADCollard) [reply scripts/exglobal_atmos_analysis.sh:922]: @RussTreadon-NOAA @emilyhcliu Sorry, I was in a meeting.   Yes, all three instruments need to be in monitor mode to facilitate bias correction spin-up.

Comment (ADCollard) [reply scripts/exglobal_atmos_analysis.sh:922]: > 
> @ADCollard , the PR description indicates that we should see changes in `fix/global_anavinfo.l127.txt`. This file is identical between `ADCollard:release/gfsda.v16` and `NOAA-EMC:release/gfsda.v16`.
> 
> Should we also see a difference between cris-fsr_n21 weights when comparing `global_satinfo.txt` from `ADCollard:release/gfsda.v16` and `NOAA-EMC:release/gfsda.v16`?

@RussTreadon-NOAA If there are no changes for `fix/global_anavinfo.l127.txt` and `fix/global_satinfo.txt`, I must have made a mistake.  Let me double-check. (My github/Gerrit skills need some work)

**Comment by ADCollard** [comment]
@RussTreadon-NOAA @emilyhcliu The changes for  fix/global_anavinfo.l127.txt and fix/global_satinfo.txt should both be in the branch now

**Comment by RussTreadon-NOAA** [comment]
I refreshed my working copy of `ADCollard:release/gfsda.v16`.   File `global_anavinfo.txt` now contains an entry for Rcov_crisn21.  File `global_satinfo.txt` now contains different values for `error` and `ermax` for some but not all cris-fsr_n21 channels. 

The `ADCollard:release/gfsda.v16` `global_satinfo.txt` file contains one additional change.  The first line of `global_satinfo.txt` is missing the leading `!` which is found in the operational `global_satinfo.txt`

```
russ.treadon@clogin09:/lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/gfsda.v16.3.10> diff fix/global_satinfo.txt /lfs/h1/ops/prod/packages/gfs.v16.3.9
/fix/fix_gsi/global_satinfo.txt |head -4
1c1
< sensor/instr/sat      chan iuse  error  error_cld  ermax   var_b    var_pg  icld_det icloud iaerosol
---
> !sensor/instr/sat      chan iuse  error  error_cld  ermax   var_b    var_pg  icld_det icloud iaerosol
```

The leading `!` is used to indicate lines to skip over when reading info files.  For example, `radinfo.f90` contains
```
    open(lunin,file='satinfo',form='formatted')
    j=0
    nlines=0
    read1:  do
       read(lunin,100,iostat=istat) cflg,crecord
       if (istat /= 0) exit
       nlines=nlines+1
       if (cflg == '!') cycle
       read(crecord,*,iostat=istat) nusis_temp,nuchan_temp,iuse_rad_temp,&
            varch_temp,varch_cld_temp,ermax_rad_temp,b_rad_temp,pg_rad_temp,icld_det_temp
```

**Comment by RussTreadon-NOAA** [comment]
Refreshed working copy of `ADCollard:release/gfsda.v16`.   The first line of `fix/global_satinfo.txt` now contains a leading `!`

**Comment by ADCollard** [comment]
> Refreshed working copy of `ADCollard:release/gfsda.v16`. The first line of `fix/global_satinfo.txt` now contains a leading `!`

Thanks for spotting that @RussTreadon-NOAA !

**Comment by HaixiaLiu-NOAA** [comment]
After getting the fix directory to my stmp space, I checked the global_satinfo.txt, global_anavinfo.l127.txt. Both are correct. The Rcov_crisn21 is a new filed in the fix directory. All changes look good to me.

**Review by HaixiaLiu-NOAA — APPROVED** [review]
all changes are good to go.

**Comment by RussTreadon-NOAA** [comment]
@emilyhcliu , when you have a chance would you please review this PR (#636) from @ADCollard

**Comment by ADCollard** [comment]
One remaining change was required.  HIRS processing is removed as the HIRS CRTM coefficient files' version numbers are not correct (and the code changes required for NOAA-21 ATMS exposes this).  More details on this [here](https://github.com/JCSDA/crtm/pull/42#issuecomment-1681258843).  We do not currently assimilate HIRS.

**Review by emilyhcliu — APPROVED** [review]
Changes look good to me.

**Comment by RussTreadon-NOAA** [comment]
@ADCollard , we have two peer review approvals.  Do you have any more changes for `ADCollard:release/gfsda.v16`? 

Once I get the green light from you, I will merge `ADCollard:release/gfsda.v16` into `NOAA-EMC:release/gfsda.v16` and create tag `gfsda.v16.3.10`.

**Comment by ADCollard** [comment]
@RussTreadon-NOAA Thanks! Please proceed.

**Pull request merged**
